.github/workflows/code-quality.yml

```
@@ -0,0 +1,248 @@
+        with:
+          fetch-depth: 0  # Full history for better analysis
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.2'
+          extensions: mbstring, xml, ctype, iconv, intl, pdo_mysql, redis
+          tools: composer:v2, phpcs, phpmd, phpcpd, psalm
+
+      - name: Cache dependencies
+        uses: actions/cache@v3
```

actionlint (1.7.7)

the runner of "actions/cache@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)